### PR TITLE
Add EC key generation sample

### PR DESCRIPTION
**Pull request by xzr**
NOTE: This example requires the asn1crypto module to be installed

### CODE REVIEW

**Comment by LudovicRousseau** [comment]
Nice.
Does your sample code works with SoftHSMv2? https://github.com/opendnssec/SoftHSMv2

**Comment by xzr** [comment]
Yes, tested on softhsm2

**Comment by xzr** [comment]
I'm still working on my application patch with sign/verify. Will add a sample later. Can do it in this pr or another one. Whichever you prefer

**Comment by LudovicRousseau** [comment]
It is fine with this PR. Add the use sample in a new commit for example.

**Comment by LudovicRousseau** [comment]
Hello,
Any news about a sample to sign/verify using ECC?

**Comment by xzr** [comment]
Hi. I had a busy work week. I will submit at the start of next week

File: samples/ec_generate.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+session.login("1234")
+
+key_id = (0x22,)
+label = 'test'
+
+# Select the curve to be used for the keys
+curve = 'secp256r1'
+
+# Setup the domain parameters, unicode conversion needed for the curve string
+domain_params = ECDomainParameters(name='named', value=NamedCurve.unmap(unicode(curve)))
```
Comment (LudovicRousseau) [inline samples/ec_generate.py]: Why do you use NamedCurve.unmap() and not NamedCurve() only?

The following code works for Python 2 and Python 3:
curve = u'secp256r1'
value=NamedCurve(curve)

**Comment by LudovicRousseau** [comment]
I used a sample code in https://github.com/LudovicRousseau/PyKCS11/blob/devel/test/test_asymetric_ECC.py#L88

I imagine you can do better.

**Comment by LudovicRousseau** [comment]
Any progress on a sign/verify sample code for EC?

**Comment by xzr** [comment]
I have a fully working solution in my application. I'll try to sit down and throw in some samples this week. Sorry, work has been crazy and been traveling a lot :)

**Comment by xzr** [comment]
sign/verify sample in separate PR: https://github.com/LudovicRousseau/PyKCS11/pull/55

**Comment by LudovicRousseau** [comment]
Committed in 9ea89f7c14c58eb35d8a2b78765bc040242a154b

Thanks

**Pull request closed without merging**